# This patch fixes several missing reiserfs_write_unlock() calls on 
# error paths not introduced by reiserfs-io-error-handling.diff
#
# jeffm@suse.com
#
diff -rup linux-2.6.8/fs/reiserfs/namei.c linux-2.6.8.fix/fs/reiserfs/namei.c
--- linux-2.6.8/fs/reiserfs/namei.c	2004-09-13 14:06:42.000000000 -0400
+++ linux-2.6.8.fix/fs/reiserfs/namei.c	2004-09-13 16:37:20.101499264 -0400
@@ -341,6 +341,7 @@ static struct dentry * reiserfs_lookup (
             REISERFS_SB(dir->i_sb)->priv_root &&
             REISERFS_SB(dir->i_sb)->priv_root->d_inode &&
 	    de.de_objectid == le32_to_cpu (INODE_PKEY(REISERFS_SB(dir->i_sb)->priv_root->d_inode)->k_objectid)) {
+          reiserfs_write_unlock (dir->i_sb);
 	  return ERR_PTR (-EACCES);
 	}
 
@@ -1091,6 +1092,7 @@ static int reiserfs_link (struct dentry 
 	return -EMLINK;
     }
     if (inode->i_nlink == 0) {
+        reiserfs_write_unlock(dir->i_sb);
         return -ENOENT;
     }
 
